src/System.Net.Http/src/System/Net/Http/Managed/AuthenticationHelper.Digest.cs

```diff
@@ -55,7 +58,15 @@ internal partial class AuthenticationHelper
             if (string.IsNullOrEmpty(parameter))
                 return false;
 
-            request.Headers.Authorization = new AuthenticationHeaderValue(Digest, parameter);
+            if (authHeader == HttpKnownHeaderNames.Authorization)
```

Comment:
Then TrySetDigestAuthToken will be returning string, and there's code duplication in checks at calling areas. Can refactor that way also.